### PR TITLE
SEO: Do not follow links on filter options

### PR DESCRIPTION
**Pull request by paveq**
### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/130

### Description (*)
This PR adds rel="nofollow" tag to filter option links and swatches links. We don't want bots following these links, as there's almost infinite amount of combinations filters can be set on large category page. All these pages have canonical tag pointing back to parent page anyway and won't be indexed as separate pages.

### Fixed Issues (if relevant)
Fixes issue where Googlebot hammers filter option links, causing lot of load on the server, and Googlebot using it's indexing quota on filtering links.

### Manual testing scenarios (*)
1. Open category page with products having filter options
2. Check that options have rel="nofollow"

### CODE REVIEW

**Comment by paveq** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @paveq. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @paveq, here is your [new Magento instance](https://pr-26213.instances.magento-community.engineering).
Admin access: [https://pr-26213.instances.magento-community.engineering/admin](https://pr-26213.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by lbajsarowicz** [comment]
Great idea, however I wonder if it's not better to make it configurable. Now it's nasty hardcode, it would be better to have an option to turn it on / off, extend --- eg. change rel="" in general to different value.

Maybe method `getRel()` on link filter object?

**Comment by paveq** [comment]
> Great idea, however I wonder if it's not better to make it configurable. Now it's nasty hardcode, it would be better to have an option to turn it on / off, extend --- eg. change rel="" in general to different value.
> 
> Maybe method `getRel()` on link filter object?

In general there does not seem to be any reason making this configurable, as other options do not make sense for facet navigation. Adding more things that are user configurable from admin has drawbacks as well.

At this point I would suggest adding getRel() which would return hard-coded value of "nofollow".  getRel() public method would be easier to customize without need of overriding the template. 

For justification please see: https://webmasters.googleblog.com/2014/02/faceted-navigation-best-and-5-of-worst.html

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @paveq,
Changes looks good to me.
Right now integration tests fails - looks not related to your changes.
Could you fix static tests or add these view files to ignore?

**Review by ihor-sviziev — APPROVED** [review]
Author can't / don't want to fix static tests, I believe it could be fixed during delivery.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6645](https://jira.corp.magento.com/browse/ENGCOM-6645) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

File: app/code/Magento/LayeredNavigation/view/frontend/templates/layer/filter.phtml [hunk 1]
```diff
@@ -7,35 +7,35 @@
...
-            <?php if ($filterItem->getCount() > 0) : ?>
-                <a href="<?= $block->escapeUrl($filterItem->getUrl()) ?>">
+            <?php if ($filterItem->getCount() > 0): ?>
+                <a href="<?= $block->escapeUrl($filterItem->getUrl()) ?>" rel="nofollow">
                     <?= /* @noEscape */ $filterItem->getLabel() ?>
-                    <?php if ($this->helper(\Magento\Catalog\Helper\Data::class)->shouldDisplayProductCountOnLayer()) : ?>
-                        <span class="count"><?= /* @noEscape */ (int)$filterItem->getCount() ?><span class="filter-count-label">
-                            <?php if ($filterItem->getCount() == 1) :
+                    <?php if ($viewModel->shouldDisplayProductCountOnLayer()): ?>
+                        <span class="count"><?= /* @noEscape */ (int)$filterItem->getCount() ?>
```
Comment (ihor-sviziev) [inline app/code/Magento/LayeredNavigation/view/frontend/templates/layer/filter.phtml:22]: @engcom-Golf when inside span tag adding new line or space - usually it adds visually additional space. Could you double check that we don't have regression here?
Also some exception happened, please check functional tests results

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
@magento run all tests

File: app/code/Magento/LayeredNavigation/view/frontend/templates/layer/filter.phtml (hunk 1 above)
Comment (engcom-Golf) [reply app/code/Magento/LayeredNavigation/view/frontend/templates/layer/filter.phtml:22]: ok

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6645](https://jira.corp.magento.com/browse/ENGCOM-6645) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by engcom-Golf** [comment]
Related PR https://github.com/magento/partners-magento2ee/pull/130

**Pull request merged**